### PR TITLE
[Google Drive] Add docs for Google Drive DLS

### PR DESCRIPTION
**Pull request by jedrazb**
## Closes https://github.com/elastic/enterprise-search-team/issues/5194

Add documentation about Document-level security for Google Drive connector.

This PR complements the code changes from: #1257

### CODE REVIEW

File: docs/reference/google_drive.md [hunk 1]
```diff
@@ -26,6 +26,9 @@ To get started, log into [Google Cloud Platform](cloud.google.com) and go to the
 
 3. **Create a Service Account.** In the `APIs & Services` section, click on `Credentials` and click on `Create credentials` to create a service account. Give your service account a name and a service account ID. This is like an email address and will be used to identify your service account in the future. Click `Done` to finish creating the service account.
 
+Your service account needs to have access to at least a following scope:
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:29]:
```suggestion
Your service account needs to have access to at least the following scope:
```

File: docs/reference/google_drive.md [hunk 2]
```diff
@@ -36,12 +39,29 @@ To get started, log into [Google Cloud Platform](cloud.google.com) and go to the
 
   Note: When you grant a service account access to a specific folder or shared drive in Google Drive, it's important to note that the permissions extend to all the children within that folder or drive. This means that any folders or files contained within the granted folder or drive inherit the same access privileges as the parent.
 
+
+#### Additional authentication prerequisites for document-level security
+
+To access user data on a Google Workspace domain, the service account that you created needs to be granted access by a super administrator for the domain. You can follow [official documentation](https://developers.google.com/cloud-search/docs/guides/delegation) to perform Google Workspace domain-wide delegation of authority.
+
+You need to grant following **OAuth Scopes** to your service account:
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:47]:
```suggestion
You need to grant the following **OAuth Scopes** to your service account:
```

File: docs/reference/google_drive.md [hunk 3]
```diff
@@ -36,12 +39,29 @@ To get started, log into [Google Cloud Platform](cloud.google.com) and go to the
...
+
+This step allows the connector to access user data and their group memberships in your Google Workspace organization.
+
 ## Configuration
 
 The following configuration fields need to be provided for setting up the connector:
 ##### `Google Drive service account JSON`
 The service account credentials generated from Google Cloud Platform (JSON string). Refer to the [Google Cloud documentation](https://developers.google.com/workspace/guides/create-credentials#create_credentials_for_a_service_account) for more information.
 
+##### `Enable document level security`
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:59]:
```suggestion
##### `Enable document-level security`
```
(Nit)
(just for consistency, optionally change all `document-level` to `document level`)

File: docs/reference/google_drive.md [hunk 4]
```diff
@@ -36,12 +39,29 @@ To get started, log into [Google Cloud Platform](cloud.google.com) and go to the
...
+Toggle to enable [document level security (DLS)](https://www.elastic.co/guide/en/enterprise-search/master/dls.html). DLS is supported for the Google Drive connector. When enabled, full syncs will fetch access control lists for each document and store them in the `_allow_access_control` field. When enabled, access control syncs will fetch users' access control lists and store them in a separate index.
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:60]:
```suggestion
Toggle to enable [document-level security (DLS)](https://www.elastic.co/guide/en/enterprise-search/master/dls.html). DLS is supported for the Google Drive connector. When enabled, full syncs will fetch access control lists for each document and store them in the `_allow_access_control` field. When enabled, access control syncs will fetch users' access control lists and store them in a separate index.
```

File: docs/reference/google_drive.md [hunk 5]
```diff
@@ -36,12 +39,29 @@ To get started, log into [Google Cloud Platform](cloud.google.com) and go to the
...
+
+##### `Google Workspace admin email`
+Google Workspace admin email. Required to enable [document level security (DLS)](https://www.elastic.co/guide/en/enterprise-search/master/dls.html). A service account with delegated authority can impersonate an admin user with permissions to access Google Workspace user data and their group memberships. Refer to the [Google Cloud documentation](https://support.google.com/a/answer/162106?hl=en) for more information.
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:63]:
```suggestion
Google Workspace admin email. Required to enable [document-level security (DLS)](https://www.elastic.co/guide/en/enterprise-search/master/dls.html). A service account with delegated authority can impersonate an admin user with permissions to access Google Workspace user data and their group memberships. Refer to the [Google Cloud documentation](https://support.google.com/a/answer/162106?hl=en) for more information.
```

File: docs/reference/google_drive.md [hunk 6]
```diff
@@ -117,6 +137,14 @@ It will attempt to extract the content from Google Suite documents (Google Docs,
 
 Advanced sync rules are not available for this connector in the present version. Currently filtering is controlled via ingest pipelines.
 
+## Document-level security
+
+Document-level security (DLS) enables you to restrict access to documents based on a user’s permissions. This feature is available by default for the Google Drive connector.
+
+Refer to [Document level security](https://www.elastic.co/guide/en/enterprise-search/master/dls.html) for more information.
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:144]:
```suggestion
Refer to [document-level security](https://www.elastic.co/guide/en/enterprise-search/master/dls.html) for more information.
```

File: docs/reference/google_drive.md (hunk 4 above)
Comment (navarone-feekery) [reply docs/reference/google_drive.md:60]: Also there are two effects to having this DLS enabled, but both sentences begin with `when enabled...`. I think they could be joined (e.g. something like `when enabled ___, and also ___`) or they could be a dot-point list.

**Review by navarone-feekery — COMMENTED** [review]
Very cool! I left some comments.

File: docs/reference/google_drive.md (hunk 4 above)
Comment (jedrazb) [reply docs/reference/google_drive.md:60]: Yup, I went with dot-point list to make it clearer 👍

**Review by navarone-feekery — APPROVED** [review]
LGTM!

**Review by wangch079 — COMMENTED** [review]
I know this is in `main` branch, but I think the practice is that, when linking to the official guide, we should link to `current` instead of `master`:

https://www.elastic.co/guide/en/enterprise-search/master/dls.html

https://www.elastic.co/guide/en/enterprise-search/current/dls.html

**Comment by jedrazb** [comment]
> I know this is in `main` branch, but I think the practice is that, when linking to the official guide, we should link to `current` instead of `master`:
> 
> https://www.elastic.co/guide/en/enterprise-search/master/dls.html
> 
> https://www.elastic.co/guide/en/enterprise-search/current/dls.html

@wangch079 I was considering that but the problem is that if we merge the PR now/soon the link with `current`: https://www.elastic.co/guide/en/enterprise-search/current/dls.html points to 404 page as `8.9` release is not publicly available yet

**Pull request merged**